### ISSUE TITLE
[PATCH] compile fails with gcc-4.7 on Fedora Core 17

### ISSUE THREAD

**Issue by romw** [BOINC/boinc-ng#1166]
**Reported by ChristianB on 7 Jun 42504572 17:03 UTC**
When compiling on Fedora Core 17 with gcc-4.7 the make fails at several places with something like:
```
error: 'unlink' was not declared in this scope
error: 'chdir' was not declared in this scope
error: 'getpid' was not declared in this scope
```
This can be fixed by adding
```
#include <unistd.h>
```
to the files:
```
tools/process_input_template.cpp
tools/create_work.cpp
sched/single_job_assimilator.cpp
vda/vda.cpp
vda/vdad.cpp
```

**Comment by romw** [comment]
**Commented by davea on 27 Dec 42504650 20:44 UTC**
(In [25837]) - Compile fixes for Fedora core 17.  From Christian B.  Fixes #1194.
- Fix various #include issues.

CODING STYLE LAW (minimal inclusion principle):
    If foo.cpp requires <blah.h>,
    #include <blah.h> in foo.cpp, NOT foo.h

**Comment by romw** [comment]
**Commented by romw on 28 May 42581401 23:56 UTC**
(In [25906])     - Compile fixes for Fedora core 17.  From Christian B.  Fixes #1194.
    - Fix various #include issues.

    CODING STYLE LAW (minimal inclusion principle):
        If foo.cpp requires <blah.h>,
        #include <blah.h> in foo.cpp, NOT foo.h

    various/
        various
